rust/nasl-interpreter/src/built_in_functions/ssh.rs

```diff
@@ -0,0 +1,2107 @@
+        _ => {
+            return Err(FunctionErrorKind::WrongArgument(
+                ("Invalid value for password").to_string(),
+            ))
+        }
+    };
+
+    let privatekey = match register.named("privatekey") {
+        Some(ContextType::Value(NaslValue::String(x))) => Some(x.as_str()),
+        None => None,
+        _ => {
+            return Err(FunctionErrorKind::WrongArgument(
+                ("Invalid value for privatekey").to_string(),
+            ))
+        }
+    };
+    let passphrase = match register.named("passphrase") {
+        Some(ContextType::Value(NaslValue::String(x))) => Some(x.as_str()),
+        None => None,
+        _ => {
+            return Err(FunctionErrorKind::WrongArgument(
+                ("Invalid invalid value for passphrase").to_string(),
+            ))
+        }
+    };
+
+    if password.is_none() && privatekey.is_none() && passphrase.is_none() {
+        //TODO: Get values from KB
+        return Err(FunctionErrorKind::Diagnostic(
+            format!("Invalid SSH session for SessionID {}", session_id),
+            Some(NaslValue::Null),
+        ));
+    }
+
+    let mut sessions = lock_sessions(&ctx.sessions().ssh_sessions)?;
+    match sessions
+        .iter_mut()
+        .enumerate()
+        .find(|(_i, s)| s.session_id == session_id)
+    {
+        Some((_i, session)) => {
+            if !session.user_set {
+                set_opt_user(session, login, session_id)?;
+            }
+            let verbose = session.verbose > 0;
+
+            // Get the authentication methods only once per session.
+            let methods: AuthMethods = {
+                if !session.authmethods_valid {
+                    get_authmethods(session, session_id, ctx.logger())?
+                } else {
+                    session.authmethods
+                }
+            };
+            if verbose {
+                ctx.logger()
+                    .info(&format!("Available methods:\n{:?}", methods));
+            }
+
+            if methods == AuthMethods::NONE {
+                return Ok(NaslValue::Number(0));
+            }
+
+            /* Check whether a password has been given.  If so, try to
+            authenticate using that password.  Note that the OpenSSH client
+            uses a different order it first tries the public key and then the
+            password.  However, the old NASL SSH protocol implementation tries
+            the password before the public key authentication.  Because we
+            want to be compatible, we do it in that order. */
+            if password.is_some() && methods.contains(AuthMethods::PASSWORD) {
+                match session.session.userauth_password(None, password) {
+                    Ok(AuthStatus::Success) => {
+                        return Ok(NaslValue::Number(0));
+                    }
+                    Ok(_) => {
+                        if verbose {
+                            ctx.logger().info(&format!(
```

Comment:
because this is use an ENV variable to switch it on, different from the general debug level. It will only be available if it is set via, which also turns the libssh debug log on     
```
let verbose = env::var("OPENVAS_LIBSSH_DEBUG")